keps/sig-node/5283-dra-resourceslice-status-device-health/README.md

```diff
@@ -0,0 +1,1017 @@
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+With Dynamic Resource Allocation (DRA), devices that can be allocated to Pods
+are advertised in ResourceSlices. Device entries in a ResourceSlice describe
+properties like identifiers, capacity, and firmware or kernel driver versions.
+
+Sometimes devices fail and an automated agent or cluster administrator must take
+some action to restore the device to a healthy state. This KEP defines a
+standard way to determine whether or not a device is considered healthy.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Since the DRA APIs currently do not expose a standard way to classify device
+health, cluster administrators are tasked with finding other means of doing that
+for each kind of device that is defined in a ResourceSlice in order to identify
+and restore unhealthy devices back to their full functionality.
+
+When unhealthy devices are no longer able to support new or existing workloads,
+DRA drivers and cluster administrators already have the option to
+[taint](https://kep.k8s.io/5055) the devices. A standard representation of
+device health in the ResourceSlice API is needed to express the state of the
+devices in cases where the side effects (`NoSchedule`/`NoExecute`) of taints are
+not desired.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Expose well-formed data in the ResourceSlice API representing the real-time
+  health of the devices a ResourceSlice describes.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Define what constitutes a "healthy" or "unhealthy" device. That distinction is
+  made by each DRA driver.
```

Comment:
This non-goal collides with what you said on the `summary` section, line 184

Reply:
This non-goal is only saying that Kubernetes doesn't care about the underlying characteristics of a device that cause a driver to consider it healthy or not. The summary says cluster administrators are interested in identifying and remediating unhealthy devices. Are those at odds with each other?

Reply:
I agree that this is problematic. Either this KEP limits itself to just defining fields that can be used in a vendor-specific way or it attaches additional semantic to those fields which then must be followed by all drivers. There are pros and cons for both.

Reply:
Based on these non-goals, the KEP seems to be in the "no semantic" camp. That raises the question whether device attributes would be sufficient, perhaps combined with ["admin-controlled device attributes"](https://github.com/kubernetes/enhancements/issues/5027).

Reply:
At first glance I don't see the connection. Device health needs to be reported, whether a device is allocated or note. The ResourceClaim status is about providing additional information related to the allocation (for example, the assigned IP in the case of a network interface). I would not use it for the overall device health, although the "raw data" escape hatch in the status would allow that.